conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -1321,6 +1324,197 @@ TEST_F(
   }
 }
 
+TEST_F(
+    zetMetricStreamerTest,
+    GivenValidMetricGroupWhenTimerBasedStreamerIsCreatedThenExpectStreamerToNotifyEventAtProperTimeAndSucceed) {
```

Comment:
How is this test different from the existing marker ones? Meaning, if this has a more accurate implementation but tests the same feature, then it may be better to remove to old one(s) .

Reply:
This test is focusing on testing whether events are getting generated at the correct time based of the sampling periods and notifyNReport values. I don't think there is any other test doing this today in a similar manner using loops and sleeps. Please let me know if otherwise.

Reply:
Events in other tests mostly have an else path as well as an escape, so no hard check like this new test.

Reply:
Not Marker tests. My confusion. 
However, my point is that most (if not all) streamer tests are using NotifyOnNevents in an incorrect way. So if this test is specific for notifyOnNReports and uses the correct logic, then would make sense to remove notifyOnNReports  from all other streamer tests.

Reply:
yes there is a JIRA to look into this https://jira.devtools.intel.com/browse/VLCLJ-2348. We can take this up as part of this since it will touch many tests.
Existing tests might be calling the `zeEventQueryStatus` for continuity of the metrics API flow from the programming guide, but maybe did not intend to explicitly check the event feature as done by the new tests added here.

Reply:
>My proposal was to simply wipe out NotfiyOnNreports from all tests that are not explicitly testing the functionality to make tests simpler. 

Yes. It is a valid point.
During this PR, there are many improvement points identified which are applicable for most of streamer tests.
I will add them to  https://jira.devtools.intel.com/browse/VLCLJ-2348, so they are not missed.